src/main/java/hudson/maven/MavenBuild.java

```diff
@@ -837,9 +837,13 @@ protected Result doRun(BuildListener listener) throws Exception {
                         LOGGER.fine( "using maven 3 " + mavenVersion );
                         factory = new Maven32ProcessFactory( getParent().getParent(), MavenBuild.this, launcher, envVars, getMavenOpts(listener, envVars), null );
                         break;
-                    default:
+                    case MAVEN_3_3:
                         LOGGER.fine( "using maven 3 " + mavenVersion );
                         factory = new Maven33ProcessFactory( getParent().getParent(), MavenBuild.this, launcher, envVars, getMavenOpts(listener, envVars), null );
+                        break;
+                    default:
```

Comment:
I do not block the PR by that. It would be even better to handle it in a separate PR since the potential impact is pretty high